File: app/src/main/java/com/mobile/model/Theater.java
package com.mobile.model;


import android.support.annotation.Nullable;

import org.parceler.Parcel;

import java.io.Serializable;

import io.realm.RealmObject;
import io.realm.annotations.PrimaryKey;
import io.realm.annotations.RealmClass;

@Parcel(value = Parcel.Serialization.BEAN, analyze = {Theater.class})
@RealmClass
public class Theater extends RealmObject implements Serializable {

    @PrimaryKey
    int id;




    String name;
    String address;
    String city;
    String state;
    String zip;
    double distance;
    double lon;
    double lat;
    String theaterChainName;
    String ticketType;
    String synopsis;
    int moviepassId;
    int tribuneTheaterId;

    public void setId(int id) {
        this.id = id;
    }

    public void setMoviepassId(int moviepassId) {
        this.moviepassId = moviepassId;
    }

    public void setTribuneTheaterId(int tribuneTheaterId) {
        this.tribuneTheaterId = tribuneTheaterId;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setAddress(String address) {
        this.address = address;
    }


    public void setCity(String city) {
        this.city = city;
    }

    public void setState(String state) {
        this.state = state;
    }

    public void setZip(String zip) {
        this.zip = zip;
    }

    public void setDistance(double distance) {
        this.distance = distance;
    }

    public void setLon(double lon) {
        this.lon = lon;
    }

    public void setLat(double lat) {
        this.lat = lat;
    }

    public void setTheaterChainName(String theaterChainName) {
        this.theaterChainName = theaterChainName;
    }

    public void setTicketType(String ticketType) {
        this.ticketType = ticketType;
    }

    public void setSynopsis(String synopsis) {
        this.synopsis = synopsis;
    }


    public Theater() {
    }

    @Nullable public String getName() {
        return name;
    }

    public boolean ticketTypeIsStandard() {
        return ticketType.matches("STANDARD");
    }

    public boolean ticketTypeIsETicket() {
        return ticketType.matches("E_TICKET");
    }

    public boolean ticketTypeIsSelectSeating() {
        return ticketType.matches("SELECT_SEATING");
    }

    public String getTicketType() {
        return ticketType;
    }

    public String getAddress() {
        return address;
    }

    public double getDistance() {
        return distance;
    }

    public int getMoviepassId() {
        return moviepassId;
    }

    public int getTribuneTheaterId() {
        return tribuneTheaterId;
    }

    public double getLon() {
        return lon;
    }

    public double getLat() {
        return lat;
    }

    public int getId() {
        return id;
    }

    public String getTitle() {
        return getName();
    }

    public String getCity() {
        return city;
    }

    public String getState() {
        return state;
    }

    public String getZip() {
        return zip;
    }


}

File: app/src/main/java/com/mobile/model/Theater.kt
package com.mobile.model

import android.os.Parcelable
import io.realm.RealmModel
import io.realm.RealmObject
import io.realm.annotations.PrimaryKey
import io.realm.annotations.RealmClass
import kotlinx.android.parcel.Parcelize

@Parcelize
@RealmClass
open class Theater(
        @PrimaryKey
        var id: Int = 0,
        var name: String? = null,
        var lat: Double = 0.0,
        var lon: Double = 0.0,
        var tribuneTheaterId: Int? = null,
        var distance:Double = 0.0,
        var city: String? = null,
        var state: String? = null,
        var address: String? = null,
        var zip: String? = null,
        var ticketType: String? = null,
        var moviepassId:Int = 0
) : RealmModel, Parcelable {

    fun ticketTypeIsStandard(): Boolean {
        return ticketType?.matches("STANDARD".toRegex()) == true
    }

    fun ticketTypeIsETicket(): Boolean {
        return ticketType?.matches("E_TICKET".toRegex()) == true
    }

    fun ticketTypeIsSelectSeating(): Boolean {
        return ticketType?.matches("SELECT_SEATING".toRegex()) == true
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as Theater

        if (id != other.id) return false
        if (name != other.name) return false
        if (lat != other.lat) return false
        if (lon != other.lon) return false
        if (tribuneTheaterId != other.tribuneTheaterId) return false
        if (distance != other.distance) return false
        if (city != other.city) return false
        if (state != other.state) return false
        if (address != other.address) return false
        if (zip != other.zip) return false
        if (ticketType != other.ticketType) return false
        if (moviepassId != other.moviepassId) return false

        return true
    }

    override fun hashCode(): Int {
        var result = id
        result = 31 * result + (name?.hashCode() ?: 0)
        result = 31 * result + lat.hashCode()
        result = 31 * result + lon.hashCode()
        result = 31 * result + (tribuneTheaterId ?: 0)
        result = 31 * result + distance.hashCode()
        result = 31 * result + (city?.hashCode() ?: 0)
        result = 31 * result + (state?.hashCode() ?: 0)
        result = 31 * result + (address?.hashCode() ?: 0)
        result = 31 * result + (zip?.hashCode() ?: 0)
        result = 31 * result + (ticketType?.hashCode() ?: 0)
        result = 31 * result + moviepassId
        return result
    }

    val cityStateZip: String?
        get() {
            return "${city}, ${state} ${zip}"
        }
}

